src/validation.h

```diff
@@ -891,16 +891,16 @@ class ChainstateManager
     void Reset();
 };
 
-extern ChainstateManager g_chainman;
+extern ChainstateManager g_chainman GUARDED_BY(::cs_main);
 
 /** @returns the most-work valid chainstate. */
-CChainState& ChainstateActive();
+CChainState& ChainstateActive() EXCLUSIVE_LOCKS_REQUIRED(::cs_main);
```

Comment:
Fixed by taking your suggestion. I didn't initially do this because I thought it created some kind of lock annotation problem, but after trying this change and rebasing the parent PR onto it, clang threw no warnings. Subsequently the diff has contracted a bit - thanks!